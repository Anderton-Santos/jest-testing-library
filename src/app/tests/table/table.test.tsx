//npx jest --watch src/app/tests/table

import { render, screen, fireEvent } from "@testing-library/react"
import Tables from "./table"

describe("Tables component", () => {
  const mockTabs = [
    { label: "Home", content: "Bem vindo a pagina home!" },
    { label: "Sobre", content: "Apresentação sobre os dados referidos" },
    { label: "main", content: "Conteudo principal da pagina" },
    { label: "Footer", content: "Parte destinada aos contantos e docs" },
  ]

  it("renders all tab buttons and the default active content", () => {
    render(<Tables tables={mockTabs} />)

    mockTabs.forEach((item) => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    })

    expect(screen.getByText("Bem vindo a pagina home!")).toBeInTheDocument();
  })

  it("changes content and style when tab button is clicked", () => {
    render(<Tables tables={mockTabs} />)

    const clickButton = screen.getByText("Sobre")
    fireEvent.click(clickButton)
    expect(clickButton).toHaveStyle("background-color: #333")

    const contentScreen = screen.queryByText("Apresentação sobre os dados referidos")

    expect(contentScreen).toBeInTheDocument()

    expect(screen.queryByText("Bem vindo a pagina home!")).not.toBeInTheDocument();
  })
  it("rendres acessible buttons with role 'button' ", () => {
    render(<Tables tables={mockTabs} />)

    const buttons = screen.getAllByRole("button")
    expect(buttons).toHaveLength(mockTabs.length)

    mockTabs.forEach((tab) => {
      expect(screen.getByRole("button", { name: tab.label })).toBeInTheDocument();
    })
  })


  it("matches snapshot", () => {
    const { asFragment } = render(<Tables tables={mockTabs} />);
    expect(asFragment()).toMatchSnapshot();
  });


})
