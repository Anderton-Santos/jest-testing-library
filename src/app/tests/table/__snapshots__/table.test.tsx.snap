// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Tables component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex"
  >
    <div>
      <button
        style="margin: 2px; padding: 0.5rem 1rem; background-color: rgb(51, 51, 51); color: rgb(255, 255, 255); border-radius: 5px; cursor: pointer;"
      >
        Home
      </button>
    </div>
    <div>
      <button
        style="margin: 2px; padding: 0.5rem 1rem; background-color: rgb(204, 204, 204); color: rgb(0, 0, 0); border-radius: 5px; cursor: pointer;"
      >
        Sobre
      </button>
    </div>
    <div>
      <button
        style="margin: 2px; padding: 0.5rem 1rem; background-color: rgb(204, 204, 204); color: rgb(0, 0, 0); border-radius: 5px; cursor: pointer;"
      >
        main
      </button>
    </div>
    <div>
      <button
        style="margin: 2px; padding: 0.5rem 1rem; background-color: rgb(204, 204, 204); color: rgb(0, 0, 0); border-radius: 5px; cursor: pointer;"
      >
        Footer
      </button>
    </div>
  </div>
  <section
    class="flex flex-col"
  >
    Bem vindo a pagina home!
  </section>
</DocumentFragment>
`;
